Sources/Sentry/SentrySpotlightTransport.m

```diff
@@ -0,0 +1,110 @@
+#import "SentrySpotlightTransport.h"
+#import "SentryDispatchQueueWrapper.h"
+#import "SentryEnvelope.h"
+#import "SentryEnvelopeItemHeader.h"
+#import "SentryEnvelopeItemType.h"
+#import "SentryLog.h"
+#import "SentryNSURLRequest.h"
+#import "SentryNSURLRequestBuilder.h"
+#import "SentryOptions.h"
+#import "SentrySerialization.h"
+#import "SentryTransport.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface
+SentrySpotlightTransport ()
+
+@property (nonatomic, strong) id<SentryRequestManager> requestManager;
+@property (nonatomic, strong) SentryNSURLRequestBuilder *requestBuilder;
+@property (nonatomic, strong) SentryOptions *options;
+@property (nonatomic, strong) SentryDispatchQueueWrapper *dispatchQueue;
+@property (nonatomic, strong, nullable) NSURL *apiURL;
+
+@end
+
+@implementation SentrySpotlightTransport
+
+- (id)initWithOptions:(SentryOptions *)options
+          requestManager:(id<SentryRequestManager>)requestManager
+          requestBuilder:(SentryNSURLRequestBuilder *)requestBuilder
+    dispatchQueueWrapper:(SentryDispatchQueueWrapper *)dispatchQueueWrapper
+{
+
+    if (self = [super init]) {
+        self.options = options;
+        self.requestManager = requestManager;
+        self.requestBuilder = requestBuilder;
+        self.dispatchQueue = dispatchQueueWrapper;
+        self.apiURL = [[NSURL alloc] initWithString:options.spotlightUrl];
+    }
+
+    return self;
+}
+
+- (void)sendEnvelope:(SentryEnvelope *)envelope
+{
+    if (self.apiURL == nil) {
+        SENTRY_LOG_WARN(@"Malformed Spotlight URL passed from the options. Not sending envelope to "
+                        @"Spotlight with URL:%@",
+            self.options.spotlightUrl);
+        return;
+    }
+
+    [self.dispatchQueue dispatchAsyncWithBlock:^{
```

Comment:
l: I see no reason to dispatch this asynchronously. The request itself will be asynchronous; everything else in here is simple behavior.